File: week1-array-string/problem01-배열두배만들기/heejae.java
// 문제: 배열 두 배 만들기
// 접근 방식: 동일한 길이의 배열을 생성한 후, 기존 원소를 하나씩 돌면서 해당 값에 두배하여 새로운 배열에 넣어준다.
// 시간복잡도: O(n)

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = new int[numbers.length];

        for(int i=0; i<numbers.length; i++) {
            answer[i] = numbers[i] * 2;
        }
        
        return answer;
    }
}

File: week2-math/problem05-약수의합/heejae.cpp
// 문제: 약수의 합
// 접근 방식: 1부터 n까지 숫자들에 대하여 n을 해당 숫자로 나눈 나머지가 0이라면 answer에 더한다.
// 시간복잡도: O(n)

#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    for(int i=1; i<=n; i++) {
        if(n % i == 0) answer+= i;
    }
    
    return answer;
}
